flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/sinks/TableSinkUtils.scala

```diff
@@ -50,21 +50,24 @@ object TableSinkUtils {
     val srcFieldTypes = query.getTableSchema.getFieldDataTypes
     val sinkFieldTypes = sink.getTableSchema.getFieldDataTypes
 
-    if (srcFieldTypes.length != sinkFieldTypes.length ||
-      srcFieldTypes.zip(sinkFieldTypes).exists { case (srcF, snkF) =>
-        !PlannerTypeUtils.isInteroperable(
-          fromDataTypeToLogicalType(srcF), fromDataTypeToLogicalType(snkF))
+    val srcLogicalTypes = srcFieldTypes.map(t => fromDataTypeToLogicalType(t))
+    val sinkLogicalTypes = sinkFieldTypes.map(t => fromDataTypeToLogicalType(t))
+
+    if (srcLogicalTypes.length != sinkLogicalTypes.length ||
+      srcLogicalTypes.zip(sinkLogicalTypes).exists {
+        case (srcType, sinkType) =>
+        !PlannerTypeUtils.isInteroperable(srcType, sinkType)
```

Comment:
Indentation?

Reply:
It matches check style,but  I’ll  keep Indentation.

Reply:
```
      srcLogicalTypes.zip(sinkLogicalTypes).exists {
        case (srcType, sinkType) =>
          !PlannerTypeUtils.isInteroperable(srcType, sinkType)
```
 or
```
      srcLogicalTypes.zip(sinkLogicalTypes).exists { case (srcType, sinkType) =>
        !PlannerTypeUtils.isInteroperable(srcType, sinkType)
```